solidity/contracts/Auth.sol

```diff
@@ -0,0 +1,148 @@
+// contracts/Auth.sol
+// SPDX-License-Identifier: MIT
```

Comment:
@tamirms @nikhilsaraf Could we use Apache 2.0? That's the license legal typically asks us to use on our code and repos.